#! /usr/bin/env python
# Forearm camera configuration

PACKAGE='hanse_gamepad'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#       Name                    Type      Reconfiguration level             Description                                                                                      Default    Min   Max
gen.add("server_hosts",         str_t,    SensorLevels.RECONFIGURE_RUNNING,    "comma separated search list of hostnames for server",                   "")
gen.add("server_port",          int_t,    SensorLevels.RECONFIGURE_RUNNING,    "ip port of server",                   8080)
gen.add("frequency",             int_t,    SensorLevels.RECONFIGURE_RUNNING,    "how often to send data to server",     1)
gen.add("joy_gate",             double_t,    SensorLevels.RECONFIGURE_RUNNING,    "min raw joystick value to be considered != 0",     1)
gen.add("scale_linear",           double_t,    SensorLevels.RECONFIGURE_RUNNING,    "scale factor for linear speed",                   1)
gen.add("scale_angular",          double_t,    SensorLevels.RECONFIGURE_RUNNING,    "scale factor for angular speed",                   1)
gen.add("scale_depth",            double_t,    SensorLevels.RECONFIGURE_RUNNING,    "scale factor for depth speed",                   1)
gen.add("axis_linear",          int_t,    SensorLevels.RECONFIGURE_RUNNING,    "js axis for linear speed",                   1)
gen.add("axis_angular",          int_t,    SensorLevels.RECONFIGURE_RUNNING,    "js axis for angular speed",                   1)
gen.add("axis_depth",          int_t,    SensorLevels.RECONFIGURE_RUNNING,    "js axis for depth speed",                   1)
gen.add("use_throttle",          bool_t,    SensorLevels.RECONFIGURE_RUNNING,    "use throttle instead of up/down buttons")
gen.add("depth_delta",          int_t,    SensorLevels.RECONFIGURE_RUNNING,    "how to go up/down in each step",                   1)
gen.add("button_down",          int_t,    SensorLevels.RECONFIGURE_RUNNING,    "button to go down",                   1)
gen.add("button_up",          int_t,    SensorLevels.RECONFIGURE_RUNNING,    "button to go up",                   1)
gen.add("button_emegency_stop",    int_t,    SensorLevels.RECONFIGURE_RUNNING,    "js button for emergency stop",                   1)
gen.add("button_hand_control",    int_t,    SensorLevels.RECONFIGURE_RUNNING,    "js button to enable hand control",                   1)

exit(gen.generate(PACKAGE, "gamepad_node", "Node"))
